config/software/ruby.rb

```diff
@@ -54,6 +54,13 @@
 
 source url: "https://cache.ruby-lang.org/pub/ruby/#{version.match(/^(\d+\.\d+)/)[0]}/ruby-#{version}.tar.gz"
 
+semver = version.match(/(\d*)\.*(\d*)\.*(\d*)\.*/)
```

Comment:
Instead of a regex here you can just use the built in gem version parsing

```
irb(main):002:0> Gem::Version.create('2.6.0').canonical_segments
=> [2, 6]
```